File: frontend/app/src/main/java/com/example/geoquiz_frontend/Enums/GameMode.java
package com.example.geoquiz_frontend.Enums;

public enum GameMode {
    CAPITALS(1, "Столицы"),
    FLAGS(2, "Флаги"),
    OUTLINES(3, "Контуры"),
    LANGUAGES(4, "Языки");

    private final int value;
    private final String displayName;

    GameMode(int value, String displayName) {
        this.value = value;
        this.displayName = displayName;
    }

    public int getValue() { return value; }
    public String getDisplayName() { return displayName; }

    public static GameMode fromValue(int value) {
        for (GameMode mode : values()) {
            if (mode.value == value) return mode;
        }
        return CAPITALS;
    }
}

File: backend/GeoQuiz_backend/GeoQuiz_backend/Domain/Enums/GameMode.cs
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace GeoQuiz_backend.Domain.Enums
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum GameMode
    {
        [EnumMember(Value = "capitals")]
        Capital = 1,
        [EnumMember(Value = "flags")]
        Flag = 2,
        [EnumMember(Value = "outlines")]
        Outline = 3,
        [EnumMember(Value = "languages")]
        Language = 4
    }
}
